src/dialogs/screenshot/screenshot.vala

```diff
@@ -0,0 +1,1322 @@
+			windowstate.screenshot_settings.set_int("delay", 0);
+		}
+	}
+
+	[DBus (name = "org.buddiesofbudgie.BudgieScreenshot")]
+	public interface ScreenshotClient : GLib.Object {
+		public abstract async void ScreenshotArea(int x, int y, int width, int height, bool include_cursor,	bool flash, string filename,
+			out bool success, out string filename_used) throws Error;
+		public abstract async void Screenshot(bool include_cursor, bool flash, string filename, out bool success, out string filename_used) throws Error;
+		public abstract async void ScreenshotWindow(bool include_frame, bool include_cursor, bool flash, string filename,
+			out bool success, out string filename_used) throws Error;
+		public abstract bool SupportScreenshotWindow() throws Error;
+	}
+
+	class MakeScreenshot {
+		int delay;
+		int scale;
+		int[]? area;
+		string screenshot_mode;
+		bool include_cursor;
+		bool include_frame;
+		bool emit_capture_sound;
+
+		CurrentState windowstate;
+		static ScreenshotClient? client = null;
+
+		private static void MakeClientConnection() {
+			if (client == null) {
+				try {
+					client = GLib.Bus.get_proxy_sync(BusType.SESSION, "org.buddiesofbudgie.BudgieScreenshot", "/org/buddiesofbudgie/Screenshot");
+				} catch (Error e) {
+					warning("MakeScreenshot get_proxy_sync %s\n", e.message);
+					client = null;
+				}
+			}
+		}
+
+		public static bool ScreenshotWindowSupport() {
+			MakeClientConnection();
+
+			try {
+				return client.SupportScreenshotWindow();
+			}
+			catch {
+				warning("Unknown Screenshot Window support");
+				return false;
+			}
+		}
+
+		public MakeScreenshot(int[]? area) {
+			MakeClientConnection();
+
+			windowstate = new CurrentState();
+			this.area = area;
+			scale = get_scaling();
+			delay = windowstate.screenshot_settings.get_int("delay");
+			emit_capture_sound = windowstate.screenshot_settings.get_boolean("screenshot-capture-sound");
+			screenshot_mode = windowstate.screenshot_settings.get_string("screenshot-mode");
+			include_cursor = windowstate.screenshot_settings.get_boolean("include-cursor");
+			include_frame = windowstate.screenshot_settings.get_boolean("include-frame");
+
+			switch (screenshot_mode) {
+				case "Selection":
+					Timeout.add(200 + (delay * 1000), () => {
+						shoot_area.begin();
+						return false;
+					});
+					break;
+				case "Screen":
+					Timeout.add(200 + (delay * 1000), () => {
+						shoot_screen.begin();
+						return false;
+					});
+					break;
+				case "Window":
+					Timeout.add(200 + (delay * 1000), () => {
+						shoot_window.begin();
+						return false;
+					});
+					break;
+			}
+		}
+
+		async void shoot_window() {
+			bool success = false;
+			string filename_used = "";
+			play_shuttersound(200);
+
+			try {
+				yield client.ScreenshotWindow(include_frame, include_cursor, true, windowstate.tempfile_path, out success, out filename_used);
+			} catch (Error e) {
+				warning("show_window %s, failed to make screenshot\n", e.message);
```

Comment:
Could we send a notification to the user that the screenshot failed?

Reply:
ordinarily yes - but these are kind of debug messages - not really user facing messages.  Plus you then have the trouble of translations.  Messages and notifications should really mean something to a user indicating what they then have to-do.  here - well nothing really, its just failed...

Reply:
Sure, but it's not really great UX when a user is expecting something to happen, and nothing does. If a screenshot fails, there is no indication about what happened, even that it failed.

In any case, maybe we should call that out of the scope of this PR.

Reply:
fair enough - think for the future.